modules/service/src/main/scala/lucuma/odb/graphql/mapping/GmosLongSlitMapping.scala

```diff
@@ -144,17 +153,20 @@ trait GmosLongSlitMapping[F[_]]
 
         common.xBin,
 
-        // TODO: a function of FPU, target SourceProfile, and ImageQuality
-        FieldRef[GmosNorthFpu]("fpu")
-          .as(
-            "defaultXBin",
-            fpu => GmosLongSlitMath.xbinNorth(
-              fpu,
-              /*placeholder*/ SourceProfile.Point(SpectralDefinition.BandNormalized(UnnormalizedSED.StellarLibrary(StellarLibrarySpectrum.O5V).some, SortedMap.empty)),
-              /*placeholder*/ ImageQuality.TwoPointZero,  // so the FPU size is the effective slit width for now
-              PosDouble.unsafeFrom(2.0)
-            )
-          ),
+        CursorField[GmosXBinning](
+          "defaultXBin",
+          cursor =>
+            for {
+              fpu <- cursor.fieldAs[GmosNorthFpu]("fpu")
+              iq  <- cursor.fieldAs[ImageQuality]("imageQuality")
+              j   <- cursor.fieldAs[Option[Json]]("sourceProfile")
+              sp  <- j.traverse(json => Result.fromEither(json.as[SourceProfile].leftMap(_.message)))
+            } yield
+              sp.fold(GmosXBinning.Two) { sourceProfile =>  // TODO: What should the real default be if there is no target
+                GmosLongSlitMath.xbinNorth(fpu, sourceProfile, iq, PosDouble.unsafeFrom(2.0))
+              },
+          List("fpu", "imageQuality", "sourceProfile")
```

Comment:
I wanted to get this out before the Christmas break, but now I'll go back and see if I can generalize this across north/south.